ote/test/clj/ote/transit_changes/detection_test.clj

```diff
@@ -577,18 +577,129 @@
             :route-key test-utilities/route-name
             :starting-week {:beginning-of-week (test-utilities/to-local-date 2019 5 20) :end-of-week (test-utilities/to-local-date 2019 5 26)}}
            (-> (nth result 0)
-               (select-keys [:no-traffic-start-date :no-traffic-end-date :route-key :no-traffic-change
-                             :different-week :route-key :starting-week]))))
+               (select-keys test-utilities/select-keys-detect-changes-for-all-routes))))
 
     (is (= {:different-week {:beginning-of-week (test-utilities/to-local-date 2019 6 3) :end-of-week (test-utilities/to-local-date 2019 6 9)}
             :route-key test-utilities/route-name
             :starting-week {:beginning-of-week (test-utilities/to-local-date 2019 5 27) :end-of-week (test-utilities/to-local-date 2019 6 2)}
+            :no-traffic-run 1
             :no-traffic-start-date (test-utilities/to-local-date 2019 6 9)}
            (-> (nth result 1)
-               (select-keys [:no-traffic-start-date :no-traffic-end-date :route-key :no-traffic-change
-                             :different-week :route-key :starting-week]))))
+               (select-keys test-utilities/select-keys-detect-changes-for-all-routes))))
     (is (= 2 (count result)))))
 
+;;;;;; TESTS for analysing route end reporting
+
+(def data-all-routes [[test-utilities/route-name-2
+                       {:route-short-name test-utilities/route-name-2,
+                        :route-long-name test-utilities/route-name-2,
+                        :trip-headsign "",
+                        :min-date nil,
+                        :max-date nil,
+                        :route-hash-id test-utilities/route-name-2}]
+                      [test-utilities/route-name
+                       {:route-short-name "",
+                        :route-long-name test-utilities/route-name,
+                        :trip-headsign "",
+                        :min-date (detection/sql-date (test-utilities/to-local-date 2019 1 1)),
+                        :max-date (detection/sql-date (test-utilities/to-local-date 2019 7 14)),
+                        :route-hash-id test-utilities/route-name}]
+                      [test-utilities/route-name-3
+                       {:route-short-name "",
+                        :route-long-name test-utilities/route-name-3,
+                        :trip-headsign "",
+                        :min-date nil,
+                        :max-date nil,
+                        :route-hash-id test-utilities/route-name-3}]])
+
+(def change-window detection/route-end-detection-threshold)
+
+(def data-ending-route-change
+  (test-utilities/weeks (test-utilities/to-local-date 2019 5 13)
+                        (list {test-utilities/route-name ["h1" "h2" "h3" "h4" "h5" "h6" "h7"]} ;; 2019 05 13
```

Comment:
Consider using (generate-basic-data-week 9)

Reply:
Will create a new PR about this.